### PR TITLE
Refactored ResctrlStater to fix rubocop issues

### LINKED ISSUE #259: stats/kernel-selftests: resolve rubocop issues of ResctrlStater class
### Reproduce
```
$ rubocop --only Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity stats/kernel-selftests

stats/kernel-selftests:720:3: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for stat is too high. [12/7]
  def stat(line, stats) ...
  ^^^^^^^^^^^^^^^^^^^^^
stats/kernel-selftests:720:3: C: Metrics/PerceivedComplexity: Perceived complexity for stat is too high. [9/8]
  def stat(line, stats) ...
  ^^^^^^^^^^^^^^^^^^^^^
```
### Reference
https://github.com/intel/lkp-tests/wiki/Coding-Standard#rubocop

### Acceptance Test
```
$ rubocop stats/kernel-selftests # there's no issue of ResctrlStater class
$ rake spec spec=stats # pass
```

### PR DESCRIPTION
**Pull request by fongjiantan**
Closes #259

### CODE REVIEW

File: stats/kernel-selftests [hunk 1]
```diff
@@ -730,39 +730,51 @@ class ResctrlStater < Stater
...
     end
   end
+
+  def parse_cat_pass(test_script, stats)
+    # The following line is repeated 3 times
+    # Pass: Check resctrl mountpoint "/sys/fs/resctrl" exists
+    stats.add "#{@test}.#{test_script}", 'pass' unless stats.key? "#{@test}.#{test_script}"
+  end
+
+  def parse_cat_skip(test_script, stats)
```
Comment (rli9) [inline stats/kernel-selftests:755]: this method is quite similar to parse_cat_pass. Here perfer to abstract a method like
```
def add_cat_result(result, test_script, stats)
```
The caller is like
```
    when /# (Pass): Check\s+(.*)/
      # Pass: Check MBM diff within 5%
      add_cat_result($1, $2, stats)
```
This add_cat_result will be used below as well.

File: stats/kernel-selftests [hunk 2]
```diff
@@ -730,39 +730,51 @@ class ResctrlStater < Stater
...
+  def parse_cat_not_ok(test_script, stats)
+    # not ok 3 CMT: test
+    test_script = test_script.split(' ', 2)[1] if %w{1 2 3 4}.include? test_script.split(' ', 2)[0]
+
+    # The following line is repeated 3 times
+    # not ok MBM: diff within 300%
+    stats.add "#{@test}.#{test_script}", 'fail' unless stats.key? "#{@test}.#{test_script}"
+  end
+
+  def parse_cat_ok(test_script, stats)
```
Comment (rli9) [inline stats/kernel-selftests:770]: this method has duplication to parse_cat_not_ok, prefer the logic like
```
def parse_cat_result(result, test_script, stats)
   test_script = test_script.split(' ', 2)[1] if %w{1 2 3 4}.include? test_script.split(' ', 2)[0]
 
  add_cat_result(result, test_script, stats)
```
The caller is like
    when /^(ok)\s+(.*)/
      # ok CAT: test
      # ok writing benchmark parameters to resctrl FS
      parse_cat_result($1, $2, stats)
```

**Comment by fongjiantan** [comment]
@rli9 The code has been updated. If you have no issues, please run the workflow. Thanks!

File: stats/kernel-selftests [hunk 3]
```diff
@@ -730,37 +730,37 @@ class ResctrlStater < Stater
...
       # ok 4 # SKIP Hardware does not support CAT or CAT is disabled
-      stats.add "#{@test}.#{@test_script}", 'skip' unless stats.key? "#{@test}.#{@test_script}"
-    when /^not ok\s+(.*)/
-      # not ok CAT: cache miss rate within 4%
-      @test_script = $1
-
-      # not ok 3 CMT: test
-      @test_script = $1.split(' ', 2)[1] if %w{1 2 3 4}.include? $1.split(' ', 2)[0]
+      add_cat_results('skip', @test_script, stats)
+    when /^\b(.*ok)\s+(.*)/
```
Comment (rli9) [inline stats/kernel-selftests:740]: thanks, what is \b used for here? Can it be more explicit to match like /^(not ok|ok)\s+(.*)/?

Comment (seowwj) [reply stats/kernel-selftests:740]: the \b was used for the word boundary until `ok`, but yes, we can change it to be more explicit 😄

**Pull request merged**